From 1ace8d70d248bc823661933f4b24c2b0716df5b0 Mon Sep 17 00:00:00 2001
From: Evoke Zhang <evoke.zhang@amlogic.com>
Date: Wed, 22 Jul 2015 16:41:31 +0800
Subject: [PATCH 5724/5965] PD#109800: defconfig: g9tv,g9bb: enable tvout &
 lcd_tablet config

CONFIG_AML_TVOUT=y
CONFIG_AML_LCD_TABLET=y
CONFIG_AML_GAMMA_DEBUG=y

Change-Id: I592c9db4ca625541912ad441e6632666cfb5cfa4
---
 arch/arm/configs/mesong9bb_defconfig     |  3 +++
 arch/arm/configs/mesong9tv_defconfig     |  3 +++
 arch/arm/mach-mesong9bb/Makefile         |  2 +-
 arch/arm/mach-mesong9tv/Makefile         |  2 +-
 drivers/amlogic/display/lcd/aml_lcd_tv.c |  4 ++--
 drivers/amlogic/display/vout/Makefile    | 10 ++++++++--
 drivers/amlogic/display/vout/lcdoutc.c   |  4 ++--
 7 files changed, 20 insertions(+), 8 deletions(-)
 mode change 100755 => 100644 drivers/amlogic/display/vout/Makefile

diff --git a/arch/arm/configs/mesong9bb_defconfig b/arch/arm/configs/mesong9bb_defconfig
index 06d8baa91b16..6e17a5db7373 100644
--- a/arch/arm/configs/mesong9bb_defconfig
+++ b/arch/arm/configs/mesong9bb_defconfig
@@ -184,6 +184,9 @@ CONFIG_VSYNC_RDMA=y
 CONFIG_AM_VIDEO=y
 CONFIG_AM_ENCODER=y
 CONFIG_AMLOGIC_VOUT=y
+CONFIG_AML_TVOUT=y
+CONFIG_AML_LCD_TABLET=y
+CONFIG_AML_GAMMA_DEBUG=y
 CONFIG_AMLOGIC_LCD=y
 CONFIG_AML_LCD_TV=y
 CONFIG_FB_AM=y
diff --git a/arch/arm/configs/mesong9tv_defconfig b/arch/arm/configs/mesong9tv_defconfig
index 77d256721c53..405cd5e683f8 100644
--- a/arch/arm/configs/mesong9tv_defconfig
+++ b/arch/arm/configs/mesong9tv_defconfig
@@ -186,6 +186,9 @@ CONFIG_VSYNC_RDMA=y
 CONFIG_AM_VIDEO=y
 CONFIG_AM_ENCODER=y
 CONFIG_AMLOGIC_VOUT=y
+CONFIG_AML_TVOUT=y
+CONFIG_AML_LCD_TABLET=y
+CONFIG_AML_GAMMA_DEBUG=y
 CONFIG_AMLOGIC_LCD=y
 CONFIG_AML_LCD_TV=y
 CONFIG_FB_AM=y
diff --git a/arch/arm/mach-mesong9bb/Makefile b/arch/arm/mach-mesong9bb/Makefile
index 9c71ae7c2b8d..d9d3e44a5568 100644
--- a/arch/arm/mach-mesong9bb/Makefile
+++ b/arch/arm/mach-mesong9bb/Makefile
@@ -51,4 +51,4 @@ plus_sec                      := $(call as-instr,.arch_extension sec,+sec)
 AFLAGS_meson-smc.o            := -Wa,-march=armv7-a$(plus_sec)
 endif
 
-obj-$(CONFIG_AMLOGIC_VOUT) += enc_clk_config.o
+obj-$(CONFIG_AML_TVOUT) += enc_clk_config.o
diff --git a/arch/arm/mach-mesong9tv/Makefile b/arch/arm/mach-mesong9tv/Makefile
index 7bb3fb11fc21..a659eb046619 100644
--- a/arch/arm/mach-mesong9tv/Makefile
+++ b/arch/arm/mach-mesong9tv/Makefile
@@ -50,4 +50,4 @@ plus_sec			:= $(call as-instr,.arch_extension sec,+sec)
 AFLAGS_meson-smc.o		:= -Wa,-march=armv7-a$(plus_sec)
 endif
 
-obj-$(CONFIG_AMLOGIC_VOUT) += enc_clk_config.o
+obj-$(CONFIG_AML_TVOUT) += enc_clk_config.o
diff --git a/drivers/amlogic/display/lcd/aml_lcd_tv.c b/drivers/amlogic/display/lcd/aml_lcd_tv.c
index afaba13365bc..f024ec3ec801 100644
--- a/drivers/amlogic/display/lcd/aml_lcd_tv.c
+++ b/drivers/amlogic/display/lcd/aml_lcd_tv.c
@@ -504,7 +504,7 @@ static void _lcd_module_enable(void)
 	_init_display_driver(&lcd_config_dft);
 }
 
-DEFINE_MUTEX(lcd_vout_mutex);
+static DEFINE_MUTEX(lcd_vout_mutex);
 
 static const vinfo_t *lcd_get_current_info(void)
 {
@@ -731,7 +731,7 @@ static struct platform_driver lcd_driver = {
 	 .probe = lcd_probe,
 	 .remove = lcd_remove,
 	 .driver = {
-		 .name = "mesonlcd",
+		 .name = "amltvlcd",
 		 .owner = THIS_MODULE,
 #ifdef CONFIG_USE_OF
 		 .of_match_table = lcd_dt_match,
diff --git a/drivers/amlogic/display/vout/Makefile b/drivers/amlogic/display/vout/Makefile
old mode 100755
new mode 100644
index 22b9615f14b3..25e8dc58f3a3
--- a/drivers/amlogic/display/vout/Makefile
+++ b/drivers/amlogic/display/vout/Makefile
@@ -3,8 +3,14 @@ obj-y += vout_notify.o  vout_serve.o
 obj-$(CONFIG_AML_VDAC_HW_SWITCH)	+= vdac_switch.o
 
 obj-$(CONFIG_AML_TVOUT)				+= tvout.o
-tvout-objs := tvoutc.o tvconf.o enc_clk_config.o
-
+tvout-objs := tvoutc.o tvconf.o
+ifeq ($(CONFIG_ARCH_MESONG9TV),y)
+else
+ifeq ($(CONFIG_ARCH_MESONG9BB),y)
+else
+tvout-objs += enc_clk_config.o
+endif
+endif
 obj-$(CONFIG_AML_LCD_TABLET)		+= lcdoutc.o
 obj-$(CONFIG_AML_LCD_TABLET)		+= lcd_extern/
 
diff --git a/drivers/amlogic/display/vout/lcdoutc.c b/drivers/amlogic/display/vout/lcdoutc.c
index fd08aff3e2ad..5445dbbc4ff0 100644
--- a/drivers/amlogic/display/vout/lcdoutc.c
+++ b/drivers/amlogic/display/vout/lcdoutc.c
@@ -327,7 +327,7 @@ static const vinfo_t *lcd_get_current_info(void)
         return &pDev->lcd_info;
 }
 
-DEFINE_MUTEX(lcd_vout_mutex);
+static DEFINE_MUTEX(lcd_vout_mutex);
 static int lcd_set_current_vmode(vmode_t mode)
 {
     mutex_lock(&lcd_vout_mutex);
@@ -2451,7 +2451,7 @@ static struct platform_driver lcd_driver = {
 	.probe = lcd_probe,
 	.remove = lcd_remove,
 	.driver = {
-		.name = "mesonlcd",
+		.name = "amltabletlcd",
 		.owner = THIS_MODULE,
 #ifdef CONFIG_USE_OF
 		.of_match_table = lcd_dt_match,
-- 
2.19.0

